core/imagedownloader.cpp

```diff
@@ -69,27 +69,40 @@ void ImageDownloader::saveImage(QNetworkReply *reply)
 	reply->deleteLater();
 }
 
-// Fetch a picture from the given filename. If this is a non-remote filename, fetch it from disk.
-// Remote files are fetched from the net in a background thread. In such a case, the output-flag
-// "stillLoading" is set to true.
+static bool isVideoFile(const QString &filename)
+{
+	// Currently, we're very crude. Simply check if the file exists and if it
+	// is an MP4-style format.
+	QFileInfo fi(filename);
+	if (!fi.exists() && !fi.isFile())
+		return false;
+	metadata md;
+	return get_metadata(qPrintable(filename), &md) == MEDIATYPE_VIDEO;
```

Comment:
not sure if this would work for all containers out there.
might as well just assume by extension and throw the file to the external player.

Reply:
If you look at https://github.com/Subsurface-divelog/subsurface/blob/master/core/metadata.cpp#L208 we try to parse the MP4 container - so this seems like a reasonable thing to do for us? This at least doesn't blindly accept any extension.
(Of course I worry about the security implications here... reading that code it should be fairly easy to fool us and to at least crash Subsurface if not worse)

Reply:
@neolit123 is in principle right - it will parse MP4s, but not AVI. So if that fails, we can still go after extension.

But: even if we fail to identify a video as such, it will simply show a broken icon (we could refine to a format unknown icon). Clicking on it will still open the system video player.

> Of course I worry about the security implications here... reading that code it should be fairly easy to fool us and to at least crash Subsurface if not worse)

I tried to make it at most a denial-of-service, which you can do in various ways anyway. If you think this is unsafe, I'll give it a second look in the next few days.

Reply:
No, I didn't see anything that glaringly unsafe. This was more an idle comment...